db/sql/0_msar.sql

```diff
@@ -740,6 +740,67 @@ CREATE TYPE __msar.col_create_def AS (
 );
 
 
+CREATE OR REPLACE FUNCTION
+msar.get_fresh_copy_name(tab_id oid, col_id smallint) RETURNS text AS $$/*
+This function generates a name to be used for a duplicated column.
+
+Given an original column name 'abc', the resulting copies will be named 'abc <n>', where <n> is
+minimal (at least 1) subject to the restriction that 'abc <n>' is not already a column of the table
+given.
+
+Args:
+  tab_id: the table for which we'll generate a column name.
+  col_id: the original column whose name we'll use as the prefix in our copied column name.
+*/
+DECLARE
+  original_col_name text;
+  idx integer := 1;
+BEGIN
+  original_col_name := attname FROM pg_attribute WHERE attrelid=tab_id AND attnum=col_id;
+  WHILE format('%s %s', original_col_name, idx) IN (
+    SELECT attname FROM pg_attribute WHERE attrelid=tab_id
+  ) LOOP
+    idx = idx + 1;
+  END LOOP;
+  RETURN format('%s %s', original_col_name, idx);
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.get_duplicate_col_create_defs(
+  tab_id oid, col_ids smallint[], new_names text[], copy_defaults boolean
```

Comment:
Could we have each argument on its own line? I found that easy to read.

```suggestion
  tab_id oid,
  col_ids smallint[],
  new_names text[],
  copy_defaults boolean
```